tests/components/mill/test_climate.py

```diff
@@ -0,0 +1,313 @@
+"""Tests for Mill climate."""
+
+from datetime import timedelta
+from unittest.mock import AsyncMock
+
+from mill import Heater, Mill
+from mill_local import Mill as MillLocal, OperationMode
+
+from homeassistant.components.climate import HVACMode
+from homeassistant.components.mill.climate import LocalMillHeater, MillHeater
+from homeassistant.components.mill.const import DOMAIN
+from homeassistant.components.mill.coordinator import MillDataUpdateCoordinator
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+async def create_heater(
+    hass: HomeAssistant,
+) -> MillHeater:
+    """Creates a Mill (cloud) Heater ready to test functions on."""
+    entry = MockConfigEntry(domain=DOMAIN)
+
+    mill_data_connection = Mill("", "", websession=AsyncMock())
+    mill_data_connection.fetch_heater_and_sensor_data = AsyncMock(return_value=None)
+
+    coordinator = MillDataUpdateCoordinator(
```

Comment:
Don't interact with integration details like the data update coordinator in tests.

Set up the integration while patching the client library appropriately and assert the core state machine and client calls, eg after service action calls.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

See the tests of other integrations that use a coordinator.